Angular: Add Named Components Migration Guide (#8171)
DevExpress/devextreme-documentation: concepts/Common/DevExtreme CLI/30 Migrate to Named Configuration Components.md

> DevExtreme Angular v24.2 introduced named configuration components. These are strictly typed and specific to each UI component. Strict typing allows code completion and hinting systems to suggest only valid configurations and detect invalid usage. We recommend upgrading to named configuration components to improve your development experience.

Reviewer: I think we can even use smth like Note (but not in red color) for the text "We recommend upgrading to named configuration components to improve your development experience."

I think we can merge this in one note with "Starting with v25.2, a warning is displayed in the console if you use generic configuration components."

The following code snippet demonstrates named and generic configuration components:

    <!-- tab: app.component.html -->
    <!-- Named configuration component -->
    <dx-form>
        <dxi-form-item></dxi-form-item>
    </dx-form>

    <!-- Generic configuration component -->
    <dx-form>
        <dxi-item></dxi-item>
    </dx-form>

> [note] Starting with v25.2, a warning is displayed in the console if you use generic configuration components.

Reviewer: we can include a link to the warning

Reviewer: https://js.devexpress.com/React/Documentation/ApiReference/UI_Components/Errors_and_Warnings/

> To upgrade your Angular application from older generic configuration components, you can use DevExtreme CLI. The `migrate angular-config-components` command uses our `devextreme-schematics` package to perform the upgrade. Run the command in your project root:

Reviewer: I'm not sure that we should use 'generic' each time with older config components. We can once explain that they were generic and use just 'older/previous/legacy' in all the other places

Reviewer: The `migrate angular-config-components` -> The `devextreme migrate angular-config-components`

Reviewer: 'Run the command in your project root:' 
It's ok because it is the easiest way but we also have the option that can be used to set the path to the project folder (if I'm not mistaken)

We can somewhere specify that migrate command has --options and later describe these options.

Author: There is no way to run the command outside of the project unfortunately, I don't think it is possible to run any NPM command this way though.

Reviewer: if the npm package is installed globally it can be possible, but ok

>     npm devextreme migrate angular-config-components

Reviewer: maybe we should include `npm install...` command here before calling migrate command
Or we can add a comment 'If devextreme-cli is installed:'
And then 'To use this command without devextreme-cli' installation....(npx...)

Please note that it should be 'npm devextreme migrate angular-config-components' -> 'devextreme migrate angular-config-components'. 
Please ask Arman to review this guide as well

    // ===== or =====

>     npx devextreme-cli migrate angular-config-components

Reviewer: Using of this command requires using of angular schematics. maybe we should include info that it is possible to install these required packages OR they can be used via npm cache
Please ask Arman if he can provide additional info in this regards.

You can add the `--dry` argument to preview changes before upgrading:

    npm devextreme migrate angular-config-components --dry
    // ===== or =====
    npx devextreme-cli migrate angular-config-components --dry

> This command can upgrade generic configuration components in both external and inline templates. To perform inline template migration, DevExtreme CLI requires the TypeScript package. You can install this package in your project folder or in the global `node_modules`:

Reviewer: I didn't understand what you wanted to say here "This command can upgrade generic configuration components in both external and inline templates."

Author: External templates are `app.component.html`/`app.html` files, while internal templates are defined within the `@Component({})` class decorator in `app.component.ts`/`app.ts` files. External templates are the default and preferred way to define templates, though we support the less common inline option as well. I will add this clarification in brackets, similar to the following:

This command can upgrade generic configuration components in both external templates (.html files) and inline templates (defined within `@Component({})` class decorators).

    // Project install
    npm i typescript --save-dev

    // Global install
    npm i -g typescript

DevExtreme CLI can also install TypeScript temporarily to perform inline template migration. This saves the package in the NPM cache.

If your application does not use inline templates, you can specify an empty `--script-include` value ('' or []) to skip inline template migration:

    npm devextreme migrate angular-config-components --script-include=''
    // ===== or =====
    npx devextreme-cli migrate angular-config-components --script-include=''

> [tags] angular

Reviewer: We prepared the screenshots for WN 25.2 and we can use it in this topic 
<img width="960" height="407" alt="image" src="https://github.com/user-attachments/assets/b9974d66-b7df-41cb-874c-cdc24062550a" />

> DevExtreme Angular v24.2 introduced named configuration components. These are strictly typed and specific to each UI component. Strict typing allows code completion and hinting systems to suggest only valid configurations and detect invalid usage.

Reviewer 2: I think readers can be a little confused here, since you have "named components" and "UI components". 

I'm not sure if this terminology is strict. Can we maybe say "configuration components" and "UI widgets"? 

```suggestion
DevExtreme Angular v24.2 introduced named configuration components. These components are strictly typed and specific to each UI widget. Strict typing allows code completion and hinting systems to suggest only valid configurations and detect invalid usage.
```

Or maybe we can say "configuration elements" and "UI components"?

```suggestion
DevExtreme Angular v24.2 introduced named configuration elements. These elements strictly typed and specific to each UI component. Strict typing allows code completion and hinting systems to suggest only valid configurations and detect invalid usage.
```

In any event, I'm just looking for a version that would use two different nouns so that there's no confusion. Maybe you can suggest another version.

Author: Unfortunately we cannot use anything other than "component" for both. "Configuration components" is the most technically correct term, which we also use everywhere else in related docs; "UI widgets" is a term we cannot use for DevExtreme components (it's similar to something like "ASP.NET Core components"). Can we leave it as is for this one instance?

Reviewer 2: OK, it will have to do then. Maybe we can specify the relation a bit more. These are "configuration components" and they have "parent UI components" or maybe "corresponding UI components". 

```suggestion
DevExtreme Angular v24.2 introduced named configuration components. They are strictly typed and specific to their parent UI component. Strict typing allows code completion and hinting systems to suggest only valid configurations and detect invalid usage.
```

The following code snippet demonstrates named and generic configuration components:

    <!-- tab: app.component.html -->
    <!-- Named configuration component -->
    <dx-form>
        <dxi-form-item></dxi-form-item>
    </dx-form>

    <!-- Generic configuration component -->
    <dx-form>
        <dxi-item></dxi-item>
    </dx-form>

[note]

- We recommend upgrading to named configuration components to improve your development experience.

> - Starting with v25.2, a warning is displayed in the console if you use generic configuration components.

Reviewer 2: I usually try to avoid comma-separated introductory clauses like "Starting with v25.2". They introduce an extra pause, they can be repetitive (especially in What's New), and they usually can be integrated into a more straightforward sentence. 

```suggestion
- DevExtreme v25.2+ displays a warning in the console if you use generic configuration components.
```

Or maybe we don't need to mention the version number at all. We display version selectors in documentation. And the article already mentioned that "named components" were introduced in v24.2. So it goes without saying that other versions wouldn't display a warning like that. 

 ```suggestion
- DevExtreme Angular components display  a warning in the console if you use generic configuration components.
```

Author: Named components were introduced in 24.2, while the new warning will be introduced in 25.2. I think both versions are important to mention.

Could we flip the sentence around a bit to get rid of the unnecessary pause?
```suggestion
- DevExtreme displays warnings in the console if you use generic configuration components starting with v25.2.
```

Reviewer 2: Got it. Didn't notice that those were two different versions, sorry :-) 

Then I would just say "DevExtreme v25.2+"

 ```suggestion
- DevExtreme v25.2+ displays warnings in the console if you use generic configuration components.
```

[/note]

> To upgrade your Angular application to named configuration components, you can use DevExtreme CLI. The `migrate` command uses our DevExtreme Schematics package to perform the upgrade. DevExtreme CLI installs this package in the NPM cache when you run `migrate`. You can also install the package manually in your project or global `node_modules`:

Reviewer 2: Ray usually recommends that we try and find alternatives for phrases "perform something". 

  ```suggestion
To upgrade your Angular application to named configuration components, you can use the following DevExtreme CLI command: `devextreme migrate`. Note that this command requires our DevExtreme Schematics package. DevExtreme CLI installs this package in the NPM cache when you run `migrate`. You can also install the package manually to your project or global `node_modules`:
```

> To upgrade your Angular application to named configuration components, you can use the following DevExtreme CLI command: `devextreme migrate`. Note that this command requires our DevExtreme Schematics package. DevExtreme CLI installs this package in the NPM cache when you run `migrate`. You can also install the package manually to your project or global `node_modules`:

Reviewer 3: ```suggestion
To upgrade your Angular application to named configuration components, use the following DevExtreme CLI command: `devextreme migrate`. Note that this command requires our DevExtreme Schematics package. DevExtreme CLI installs this package in the NPM cache when you run `migrate`. You can also install the package to your project or global `node_modules`:
```

    // Project install
    npm i devextreme-schematics --save-dev

    // Global install
    npm i -g devextreme-schematics

> To perform the migration to named configuration components, run `devextreme migrate angular-config-components` in your project root:

Reviewer 2: I don't think we need to mention the command in text, if we then list it in the code sample. 

```suggestion
To upgrade your application to named configuration components, run the command in your project root:
```

    // If DevExtreme CLI is installed (in your project or global node_modules)
    devextreme migrate angular-config-components

    // If DevExtreme CLI is not installed
    npx devextreme-cli migrate angular-config-components

<img src="/images/DevExtremeCLI/devextreme-cli-migrate-terminal.png" alt="A terminal window that displays the output of two commands, including a DevExtreme CLI command."></img>

You can add the `--dry` argument to preview changes before upgrading:

    // If DevExtreme CLI is installed (in your project or global node_modules)
    devextreme migrate angular-config-components --dry
    
    // If DevExtreme CLI is not installed
    npx devextreme-cli migrate angular-config-components --dry

> This command can upgrade generic configuration components in both external templates (.html files) and inline templates (defined within `@Component({})` class decorators). To perform inline template migration, DevExtreme CLI requires the TypeScript package. You can install this package in your project folder or in the global `node_modules`:

Reviewer 2: ```suggestion
This command can upgrade generic configuration components in both external templates (.html files) and inline templates (defined within `@Component({})` class decorators). Inline template migration requires the TypeScript package. You can install this package to your project folder or the global `node_modules`:
```

    // Project install
    npm i typescript --save-dev

    // Global install
    npm i -g typescript

> DevExtreme CLI can also install TypeScript temporarily to perform inline template migration. This saves the package in the NPM cache.

Reviewer 2: ```suggestion
DevExtreme CLI can also install TypeScript temporarily (to the NPM cache).
```

> If your application does not use inline templates, you can specify an empty `--script-include` value ('' or []) to skip inline template migration:

Reviewer 2: You seem to state a reason for me to use the parameter - "to skip inline template migration". But since the application does not use inline templates, the output should be the same whether I skip it or not. So this is not the real reason why you want to use an extra parameter. The only real reason I can imagine is to shorten processing time. 

```suggestion
If your application does not use inline templates, you can speed up migration. Instruct the command to skip script files.  Use the  `--script-include` argument with an empty value - `''` or `[]`:
```

Author: Yeah, speeding up the process is essentially the main reason to skip inline templates. I would like to slightly rearrange/shorten your suggestion though:
```suggestion
If your application does not use inline templates, you can instruct the command to skip script files to speed up migration. Specify an empty `--script-include` value ('' or []):
```
